libraries/FFGAppImport/AssetImport/FetchContent.cs

```diff
@@ -100,31 +101,39 @@ public string fetchAppVersion()
                 {
                     return finder.ExtractObbVersion();
                 }
-                string resourcesAssets = finder.location + "/resources.assets";
+                string resourcesAssets = Path.Combine(finder.location, "resources.assets");
                 if (!File.Exists(resourcesAssets))
                 {
                     ValkyrieDebug.Log("Could not find main assets file: " + resourcesAssets);
                     return appVersion;
                 }
 
                 // We assume that the version asset is in resources.assets
-                var assetsFile = new AssetsFile(resourcesAssets, new EndianStream(File.OpenRead(resourcesAssets), EndianType.BigEndian));
-
-                // Look through all listed assets
-                foreach (var asset in assetsFile.preloadTable.Values)
+                using (var fs = File.OpenRead(resourcesAssets))
                 {
-                    // Check all text assets
-                    if (asset.Type2 == 49) //TextAsset
+                    using (var endianStream = new EndianStream(fs, EndianType.BigEndian))
                     {
-                        // Load as text
-                        var textAsset = new Unity_Studio.TextAsset(asset, false);
-                        // Check if called _version
-                        if (asset.Text.Equals("_version"))
+                        var assetsFile = new AssetsFile(resourcesAssets, endianStream);
+
+                        // Look through all listed assets
+                        foreach (var asset in assetsFile.preloadTable.Values)
                         {
-                            // Read asset content
-                            textAsset = new Unity_Studio.TextAsset(asset, true);
-                            // Extract version
-                            appVersion = System.Text.Encoding.UTF8.GetString(textAsset.m_Script);
+                            // Check all text assets
+                            if (asset.Type2 != 49) //TextAsset
+                            {
+                                continue;
+                            }
+                            // Load as text
+                            var textAsset = new Unity_Studio.TextAsset(asset, false);
+                            // Check if called _version
+                            if (asset.Text.Equals("_version"))
+                            {
+                                // Read asset content
+                                textAsset = new Unity_Studio.TextAsset(asset, true);
+                                // Extract version
+                                appVersion = System.Text.Encoding.UTF8.GetString(textAsset.m_Script);
+                                break;

```

Comment:
I added a break here to return on the first match. The previous version would continue the search. Saw no problem with that, but maybe this was intended.